osu.Game.Rulesets.Osu/Mods/OsuModRandom.cs

```diff
@@ -0,0 +1,364 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.UserInterface;
+using osu.Framework.Platform;
+using osu.Framework.Utils;
+using osu.Game.Beatmaps;
+using osu.Game.Configuration;
+using osu.Game.Graphics.UserInterface;
+using osu.Game.Overlays.Settings;
+using osu.Game.Rulesets.Mods;
+using osu.Game.Rulesets.Objects;
+using osu.Game.Rulesets.Osu.Beatmaps;
+using osu.Game.Rulesets.Osu.Objects;
+using osu.Game.Rulesets.Osu.UI;
+using osuTK;
+
+namespace osu.Game.Rulesets.Osu.Mods
+{
+    /// <summary>
+    /// Mod that randomises the positions of the <see cref="HitObject"/>s
+    /// </summary>
+    public class OsuModRandom : ModRandom, IApplicableToBeatmap
+    {
+        public override string Description => "It never gets boring!";
+        public override bool Ranked => false;
+
+        // The distances from the hit objects to the borders of the playfield they start to "turn around" and curve towards the middle.
+        // The closer the hit objects draw to the border, the sharper the turn
+        private const byte border_distance_x = 192;
+        private const byte border_distance_y = 144;
+
+        private static readonly Bindable<int> seed = new Bindable<int>
+        {
+            Default = -1
+        };
+
+        private static readonly BindableBool random_seed = new BindableBool
+        {
+            Value = true,
+            Default = true
+        };
```

Comment:
These should not be `static`. See any other mod with settings.

Reply:
If they aren't `static` the text of the `OsuNumberBox` doesn't change if the value of `seed` is changed and I can't access `random_seed.Value` from `SeedControl.Update()` (which I do with `CustomSeedDisabled` atm (line 55)), which I need to enable/disable the `OsuNumberBox`.

Reply:
The proper way to address that is to pass the mod instance down to the control, if absolutely necessary. Static properties are a huge no-no since they will be shared across *all* instances of this class, and you can end up polluting other instances this way.